app/controllers/cogy/cogy_controller.rb

```diff
@@ -20,7 +20,13 @@ def command
 
       begin
         if (command = Cogy.commands[cmd])
-          result = Context.new(command, args, opts, user, cog_env).invoke
+          result = if command.readonly
+                     ActiveRecord::Base.connected_to(role: Cogy.readonly_role) do
+                       Context.new(command, args, opts, user, cog_env).invoke
```

Comment:
In order to avoid duplication I would save Context into a variable and re-use it.
`context = Context.new(command, args, opts, user, cog_env)`

Reply:
:hbang: for the idea!